lib/graphql/object_type.rb

```diff
@@ -117,25 +127,33 @@ def normalize_interfaces(ifaces)
     end
 
     def interface_fields
-      load_interfaces
+      load_interfaces unless @clean_inherited_fields
       @clean_inherited_fields
     end
 
-    def load_interfaces
-      @clean_interfaces ||= begin
-        ensure_defined
-        clean_ifaces = normalize_interfaces(@dirty_interfaces)
-        clean_inherited_ifaces = normalize_interfaces(@dirty_inherited_interfaces)
-        inherited_fields = {}
-        clean_inherited_ifaces.each do |iface|
-          # This will be found later in schema validation:
+    def load_interfaces(ctx = GraphQL::Query::NullContext)
+      ensure_defined
```

Comment:
This is the meat of the change, a big negative is that we lose a lot of caching for interfaces.